### PR TITLE
Fix #13: Design goals for compactness and readability

### LINKED ISSUE #13: Formulate design goals w.r.t. compactness and readability
Copied from https://github.com/sarif-standard/sarif-spec/issues/126, created by @lgolding:

### PR DESCRIPTION
**Pull request by lgolding**
Affects internal documentation only (not spec text or schema).

### CODE REVIEW

File: Documents/GuidingPrinciples.md [hunk 1]
```diff
@@ -13,6 +15,26 @@
...
 
-    Example: SARIF allows you to specify the start of a `region` either with the `startLine` and `startColumn` properties, or with the `offset` property. It's ok for both to be present, as long as they're consistent.
\ No newline at end of file
+    Example: SARIF allows you to specify the start of a `region` either with the `startLine` and `startColumn` properties, or with the `offset` property. It's ok for both to be present, as long as they're consistent.
+
+9. In cases where it might help producers to provide valid property values, property names may specify their format, for example, `workItemUris`, `startTimeUtc`, and `instanceGuid`.
+
+10. In cases where the a composite property (an array or an object) can legally be empty, SARIF allows a producer to either to emit an empty array or object, or to omit the property (even though including it would make the log file larger).
```
Comment (michaelcfanning) [inline Documents/GuidingPrinciples.md]: neither #9 or #10 provide much value imo

File: Documents/GuidingPrinciples.md [hunk 2]
```diff
@@ -13,6 +15,26 @@
...
+
+# Principles related to compactness and readability
```
Comment (michaelcfanning) [inline Documents/GuidingPrinciples.md]: >Principles [](start = 2, length = 10)

These aren't bad but I think they could be improved. 'readability' as a term here also implies 'searchability' or refers to completeness, and we haven't really covered this aspect. In general, I feel like these guidelines are suffering in that some principles are very high level and other items are very targeted to a specific design approach.

All the compactness/readability issues derive from a principle that I don't see expressed here: SARIF's core design is intended to support emitting a flat list/stream of message, such as what might appears in an error list. ie, run.result is the heart of the design and is intended to be self-sufficient. augmenting this core information, we've chosen to provide supplemental tables to augment core results production for other scenarios (these scenarios include debuggability, which is why you can embed files in the files table, and compliance, which is why we support arbitrary file lists and persisting hash values).

**Comment by michaelcfanning** [comment]
This particular work item might be one that would be best for us to partner on.

**Comment by lgolding** [comment]
This is good feedback. Would you like me to try to incorporate it? Or -- taking my cue from the word "partner" -- do you have some specific ways you want to express your ideas? Would you like to pull the branch and push your proposed changes?

---
In reply to: [448027933](https://github.com/oasis-tcs/sarif-spec/pull/300#issuecomment-448027933) [](ancestors = 448027933)

**Comment by lgolding** [comment]
Sure, no hurry on this.

---
In reply to: [448050286](https://github.com/oasis-tcs/sarif-spec/pull/300#issuecomment-448050286) [](ancestors = 448050286,448028667,448027933)

**Comment by lgolding** [comment]
Well, _this_ is never happening, now, is it? :-) Closing.

**Pull request closed without merging**